tests/test_sequence.py

```diff
@@ -137,6 +138,288 @@ def test_magnetic_field():
+    sweep = Pulse.ConstantAmplitude(
+        2.3 * 2 * np.pi,
+        RampWaveform(400, -4 * np.pi, 4 * np.pi),
+        1.0,
+    )
+    fall = Pulse.ConstantDetuning(
+        RampWaveform(500, 2.3 * 2 * np.pi, 0.0),
+        4 * np.pi,
+        0.0,
+    )
+
+    # Device checkout
+    seq = Sequence(reg, Chadoq2)
+    with pytest.warns(
+        UserWarning,
+        match="Switching a sequence to the same device"
+        + " returns the sequence unchanged.",
+    ):
+        seq.switch_device(Chadoq2)
+    assert seq.switch_device(Chadoq2)._device == Chadoq2
+
+    # Test not strict mode
+    seq.declare_channel("ising", "rydberg_global")
+    assert "ising" in seq.switch_device(test_device1).declared_channels
+
+    seq = Sequence(reg, Chadoq2)
+    with pytest.raises(
+        NotImplementedError,
+        match="Switching the device of a sequence to one"
+        + " with reusable channels is not supported.",
+    ):
+        seq.switch_device(MockDevice)
+
+    seq = Sequence(reg, MockDevice)
+    seq.declare_channel("ising", "rydberg_global")
+    with pytest.raises(
+        ValueError,
+        match="Device match failed because the devices"
+        + " have different Rydberg levels.",
+    ):
+        seq.switch_device(IroiseMVP, True)
+
+    # Different Channels basis
+    test_device1.change_rydberg_level(IroiseMVP.rydberg_level)
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("digital", "raman_global", "q1")
+    with pytest.raises(
+        TypeError,
+        match="No match for channel digital with the"
+        + " right basis and addressing.",
+    ):
+        seq.switch_device(IroiseMVP)
+
+    # Different addressing channels
+    test_device1.change_rydberg_level(test_device2.rydberg_level)
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("ising", "raman_global")
+
+    with pytest.raises(
+        TypeError,
+        match="No match for channel ising with the"
+        + " right basis and addressing.",
+    ):
+        seq.switch_device(test_device2)
+
+    # Channel addressing and basis issue
+    test_device2.change_rydberg_level(IroiseMVP.rydberg_level)
+    seq = Sequence(reg, test_device2)
+    seq.declare_channel("ising", "rmn_local")
+
+    with pytest.raises(
+        TypeError,
+        match="No match for channel ising with the"
+        + " right basis and addressing.",
+    ):
+        seq.switch_device(IroiseMVP)
+
+    # Strict: Jump_phase_time & CLock-period criteria
+    # Jump_phase_time check 1: phase not nill
+    seq = Sequence(reg, test_device3)
+    twin_seq = Sequence(reg, test_device1)
+
+    seq.declare_channel("ising", channel_id="rydberg_global")
+    twin_seq.declare_channel("ising", channel_id="rydberg_global")
+    seq.add(rise, "ising")
+    twin_seq.add(rise, "ising")
+    seq.add(sweep, "ising")
+    twin_seq.add(sweep, "ising")
+    new_seq = seq.switch_device(test_device1, True)
+
+    assert sample(new_seq).__repr__() == sample(twin_seq).__repr__()
```

Comment:
This doesn't seem to be a good way to compare samples. See how it can be done properly [here](https://github.com/pasqal-io/Pulser/blob/6b7a2f1e9715d702fae7c432983b4058eaeb7d82/tests/test_sequence_sampler.py#L48-L61), for example